src/libraries/System.Private.CoreLib/src/System/Type.cs

```diff
@@ -424,7 +424,14 @@ public virtual MemberInfo GetMemberWithSameMetadataDefinitionAs(MemberInfo membe
             | DynamicallyAccessedMemberTypes.PublicNestedTypes)]
         public virtual MemberInfo[] GetDefaultMembers() => throw NotImplemented.ByDesign;
 
-        public virtual RuntimeTypeHandle TypeHandle => throw new NotSupportedException();
+        public virtual RuntimeTypeHandle TypeHandle
+        {
+#if NATIVEAOT
```

Comment:
Ugh. I thought that the native AOT special casing is not going to be needed. Do we really need it?

Reply:
It is not needed for `RuntimeTypeHandle.ToIntPtr()` but it is still needed for old pattern `typeof(int).TypeHandle.Value;` as @MichalStrehovsky has just mentioned that this pattern is used in COM so I decided to get it back. If we recommend user to always use `RuntimeTypeHandle.ToIntPtr` I guess we can remove it?

Reply:
>It is not needed for RuntimeTypeHandle.ToIntPtr() but it is still needed for old pattern typeof(int).TypeHandle.Value

`RuntimeTypeHandle.ToIntPtr` pattern is `RuntimeTypeHandle.ToIntPtr(typeof(int).TypeHandle)`. What makes it work without `get_TypeHandle` being treated as intrinsics?

In any case, could you please delete the ifdef around the attribute, and centralize the differences in the JIT? Intrinsic annotations are not meant to be 100% precise. We have number of them where the method is treated as Intrinsic on one runtime flavor, but not on the other runtime flavor. We do not encode this detail using ifdefs in .cs files. (In other words, everything should work fine if all CoreLib methods are reported to the JIT as intrinsics.)

Reply:
> typeof(int).TypeHandle.Value; as @MichalStrehovsky has just mentioned that this pattern is used in COM so I decided to get it back

The pattern in COM generators is `typeof(Foo).TypeHandle` without the Value part: https://github.com/dotnet/runtime/blob/5eef91d51a7b292027aabd7be2dce45199bb63ac/src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/ComClassGenerator.cs#L138